server/api/db/sqldb/db.py

```diff
@@ -1741,6 +1742,69 @@ def update_function(
             self._upsert(session, [function])
             return function.struct
 
+    def update_function_external_invocation_url(
+        self,
+        session,
+        name: str,
+        url: str,
+        project: str = "",
+        tag: str = "",
+        hash_key: str = "",
+        operation=Operation.ADD,
+    ):
+        """
+        This function updates the external invocation URLs of a function within a project.
+        It can add or remove URLs based on the specified `operation` which can be
+        either ADD or REMOVE of type :py:class:`~mlrun.types.Operation`
+        """
+        project = project or config.default_project
+        normalized_function_name = mlrun.utils.normalize_name(name)
+        query = self._query(
+            session, Function, name=normalized_function_name, project=project
+        )
+        uid = self._get_function_uid(
+            session=session,
+            name=normalized_function_name,
+            tag=tag or "latest",
+            hash_key=hash_key,
+            project=project,
+        )
+        if uid:
+            query = query.filter(Function.uid == uid)
```

Comment:
use `self._find_functions` instead. it returns a `query` object you can then `one_or_none`

Reply:
@TomerShor I've just tried it and for some reason this method doesn't find a correct function for me...

Reply:
That's odd.
Using `_get_function_uid` this way results with 2 db queries - first fetch the tag and get the uid, then attach the uid to the query and get the function.
`_find_functions` optimizes this by doing an outer join on the tags, so we end up with only a single query.
In large scale systems this matters. Also - It's best reusing the existing methods then implementing queries differently for different flows.

I would give it another minute to check why it doesn't yield the same results.

Reply:
@TomerShor `_find_functions` performs a request itself, so there is not way to do `one_or_none`. And also in `get_function` method we don't use _find_functions btw

Reply:
This function doesn't provide an interface for getting a specific function by it's name. This method returns all the functions which names start with given `name`, so that's why we can't use this method.